src/contracts/pods/EigenPod.sol

```diff
@@ -164,267 +175,355 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
-         * the `verifyAndProcessWithdrawal` function should be called instead.
-         */
-        if (validatorCurrentBalanceGwei == 0) {
-            uint64 slashedStatus = Endian.fromLittleEndianUint64(validatorFields[BeaconChainProofs.VALIDATOR_SLASHED_INDEX]);
-            require(slashedStatus == 1, "EigenPod.verifyOvercommittedStake: Validator must be slashed to be overcommitted");
-            //Verify the validator fields, which contain the validator's slashed status
-            BeaconChainProofs.verifyValidatorFields(
-                validatorIndex,
-                beaconStateRoot,
-                proofs.validatorFieldsProof,
-                validatorFields
-            );
-        }
         // verify ETH validator's current balance, which is stored in the `balances` container of the beacon state
-       BeaconChainProofs.verifyValidatorBalance(
+        BeaconChainProofs.verifyValidatorBalance(
             validatorIndex,
-            beaconStateRoot,
+            proofs.beaconStateRoot,
             proofs.validatorBalanceProof,
             proofs.balanceRoot
         );
+        //verify provided slot is valid against beaconStateRoot
+        BeaconChainProofs.verifySlotRoot(
+            proofs.beaconStateRoot,
+            proofs.slotRoot,
+            proofs.slotProof
+        );
+
+        uint64 currentRestakedBalanceGwei = validatorInfo.restakedBalanceGwei;
+
+        // calculate the effective (pessimistic) restaked balance
+        uint64 newRestakedBalanceGwei = _calculateRestakedBalanceGwei(validatorNewBalanceGwei);
+
+        //update the balance
+        validatorInfo.restakedBalanceGwei = newRestakedBalanceGwei;
+
+        //update the most recent balance update slot
+        validatorInfo.mostRecentBalanceUpdateSlot = Endian.fromLittleEndianUint64(proofs.slotRoot);
 
-        // mark the ETH validator as overcommitted
-        validatorStatus[validatorIndex] = VALIDATOR_STATUS.OVERCOMMITTED;
+        //record validatorInfo update in storage
+        _validatorPubkeyHashToInfo[validatorPubkeyHash] = validatorInfo;
+        
 
-        emit ValidatorOvercommitted(validatorIndex);
+        emit ValidatorBalanceUpdated(validatorIndex, newRestakedBalanceGwei);
 
-        // remove and undelegate shares in EigenLayer
-        eigenPodManager.recordOvercommittedBeaconChainETH(podOwner, beaconChainETHStrategyIndex, REQUIRED_BALANCE_WEI);
+        if (newRestakedBalanceGwei != currentRestakedBalanceGwei){
+            int256 sharesDelta = _calculateSharesDelta(newRestakedBalanceGwei * GWEI_TO_WEI, currentRestakedBalanceGwei* GWEI_TO_WEI);
+            // update shares in strategy manager
+            eigenPodManager.recordBeaconChainETHBalanceUpdate(podOwner, beaconChainETHStrategyIndex, sharesDelta);
+        }
     }
 
     /**
      * @notice This function records a full withdrawal on behalf of one of the Ethereum validators for this EigenPod
```

Comment:
this is for both partial and full withdrawals now, correct?

Reply:
fixed